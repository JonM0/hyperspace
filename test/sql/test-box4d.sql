SELECT '((0,1,1,0), (1,2,8,3))'::box4d;

SELECT '((0,5,-3,0), (1,2,8,3))'::box4d;

SELECT '((0,+inf,-3,-Inf), (1,-Inf,8,3))'::box4d;

SELECT point4d '(1,1,0,0)' <@ box4d '((0,0,0,0), (2,2,2,2))';

SELECT box4d '((0,0,0,0), (2,2,2,2))' @> point4d '(1,1,0,0)';

SELECT point4d '(1,1,0,0)' <@ box4d '((0,0,-inf,0), (2,2,2,+inf))';

SELECT point4d '(1,1,0,-1)' <@ box4d '((0,0,0,0), (2,2,2,2))';
select point4d '(1,1,3,2)' <@ box4d '((0,0,0,0), (2,2,2,2))';
