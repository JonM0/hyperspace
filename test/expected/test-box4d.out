SELECT '((0,1,1,0), (1,2,8,3))'::box4d;
         box4d         
-----------------------
 ((0,1,1,0),(1,2,8,3))
(1 row)

SELECT '((0,5,-3,0), (1,2,8,3))'::box4d;
         box4d          
------------------------
 ((0,2,-3,0),(1,5,8,3))
(1 row)

SELECT ('((1,2,3,4), (6,7,8,9))'::box4d)[0];
   box4d   
-----------
 (1,2,3,4)
(1 row)

SELECT ('((1,2,3,4), (6,7,8,9))'::box4d)[1];
   box4d   
-----------
 (6,7,8,9)
(1 row)

SELECT (('((1,2,3,4), (6,7,8,9))'::box4d)[0])[2];
 box4d 
-------
     3
(1 row)

SELECT '((0,+inf,-3,-Inf), (1,-Inf,8,3))'::box4d;
                     box4d                     
-----------------------------------------------
 ((0,-Infinity,-3,-Infinity),(1,Infinity,8,3))
(1 row)

SELECT point4d '(1,1,0,0)' <@ box4d '((0,0,0,0), (2,2,2,2))';
 ?column? 
----------
 t
(1 row)

SELECT box4d '((0,0,0,0), (2,2,2,2))' @> point4d '(1,1,0,0)';
 ?column? 
----------
 t
(1 row)

SELECT point4d '(1,1,0,0)' <@ box4d '((0,0,-inf,0), (2,2,2,+inf))';
 ?column? 
----------
 t
(1 row)

SELECT point4d '(1,1,0,-1)' <@ box4d '((0,0,0,0), (2,2,2,2))';
 ?column? 
----------
 f
(1 row)

select point4d '(1,1,3,2)' <@ box4d '((0,0,0,0), (2,2,2,2))';
 ?column? 
----------
 f
(1 row)

